test/unit/CrosschainDeployScriptTest.t.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+
+pragma solidity 0.8.20;
+
+import {Test, console} from "forge-std/Test.sol";
+import {CrosschainDeployScript} from "../../src/CrosschainDeployScript.sol";
+import {SimpleContract} from "../SimpleContract.sol";
+import {MockCrosschainDeployAdapter} from "../mocks/MockCrosschainDeployAdapter.sol";
+
+contract CrosschainDeployScriptTest is Test {
+    address crosschainDeployAdapterAddress;
+    // start the dependent contracts
+
+    function setUp() public {
+        vm.startBroadcast();
+        MockCrosschainDeployAdapter mockCrosschainDeployAdapter = new MockCrosschainDeployAdapter();
+        vm.stopBroadcast();
+        crosschainDeployAdapterAddress = address(mockCrosschainDeployAdapter);
+    }
+
+    // add a deployment target
+    function testAddDeploymentTarget() public {
+        CrosschainDeployScript crosschainDeployScript = new CrosschainDeployScript("SimpleContract.sol:SimpleContract");
+        bytes memory constructorArgs = "0x";
+        bytes memory initData = "0x";
+        crosschainDeployScript.setCrosschainDeployContractAddress(crosschainDeployAdapterAddress);
+
+        crosschainDeployScript.addDeploymentTarget("sepolia", constructorArgs, initData);
+        crosschainDeployScript.deploy{value: 2}(500, false);
```

Comment:
shouldn't you add some assertions (expectCall, check contract state) etc.

Reply:
It's not really random, test is executing in same block/tx so you know all the parameters for random salt and you can generate same one in test case. That said, there should be method for user to set salt to deal with the case when it already deployed on 3 network on the same address and now wishes to deploy on two more for which he needs to submit same salt as on initial deploy

Reply:
all good!